src/quicktimevideo.cpp

```diff
@@ -0,0 +1,1599 @@
+    io_->readOrThrow(buf.data(), 48);
+    xmpData_["Xmp.video.Software"] = Exiv2::toString(buf.data());
+    io_->readOrThrow(buf.data(), 4);
+    xmpData_["Xmp.video.ISO"] = buf.read_uint32(0, littleEndian);
+  }
+
+  io_->seek(cur_pos + size_external, BasicIo::beg);
+}  // QuickTimeVideo::CameraTagsDecoder
+
+void QuickTimeVideo::userDataDecoder(size_t size_external) {
+  size_t cur_pos = io_->tell();
+  const TagVocabulary* td;
+  const TagVocabulary *tv, *tv_internal;
+
+  const long bufMinSize = 100;
+  DataBuf buf(bufMinSize);
+  size_t size = 0, size_internal = size_external;
+  std::memset(buf.data(), 0x0, buf.size());
+
+  while ((size_internal / 4 != 0) && (size_internal > 0)) {
+    buf.data()[4] = '\0';
+    io_->readOrThrow(buf.data(), 4);
+    size = buf.read_uint32(0, bigEndian);
+    if (size > size_internal)
+      break;
+    size_internal -= size;
+    io_->readOrThrow(buf.data(), 4);
+
+    if (buf.data()[0] == 169)
+      buf.data()[0] = ' ';
+    td = find(userDatatags, Exiv2::toString(buf.data()));
+
+    tv = find(userDataReferencetags, Exiv2::toString(buf.data()));
+
+    if (size == 0 || (size - 12) <= 0)
+      break;
+
+    else if (equalsQTimeTag(buf, "DcMD") || equalsQTimeTag(buf, "NCDT"))
+      userDataDecoder(size - 8);
+
+    else if (equalsQTimeTag(buf, "NCTG"))
+      NikonTagsDecoder(size - 8);
+
+    else if (equalsQTimeTag(buf, "TAGS"))
+      CameraTagsDecoder(size - 8);
+
+    else if (equalsQTimeTag(buf, "CNCV") || equalsQTimeTag(buf, "CNFV") || equalsQTimeTag(buf, "CNMN") ||
+             equalsQTimeTag(buf, "NCHD") || equalsQTimeTag(buf, "FFMV")) {
+      io_->readOrThrow(buf.data(), size - 8);
+      xmpData_[exvGettext(tv->label_)] = Exiv2::toString(buf.data());
+    }
+
+    else if (equalsQTimeTag(buf, "CMbo") || equalsQTimeTag(buf, "Cmbo")) {
+      io_->readOrThrow(buf.data(), 2);
+      buf.data()[2] = '\0';
+      tv_internal = find(cameraByteOrderTags, Exiv2::toString(buf.data()));
+
+      if (tv_internal)
+        xmpData_[exvGettext(tv->label_)] = exvGettext(tv_internal->label_);
+      else
+        xmpData_[exvGettext(tv->label_)] = Exiv2::toString(buf.data());
+    }
+
+    else if (tv) {
+      io_->readOrThrow(buf.data(), 4);
+      io_->readOrThrow(buf.data(), size - 12);
+      xmpData_[exvGettext(tv->label_)] = Exiv2::toString(buf.data());
+    }
+
+    else if (td)
+      tagDecoder(buf, size - 8);
+  }
+
+  io_->seek(cur_pos + size_external, BasicIo::beg);
+}  // QuickTimeVideo::userDataDecoder
+
+void QuickTimeVideo::NikonTagsDecoder(size_t size_external) {
+  size_t cur_pos = io_->tell();
+  DataBuf buf(200), buf2(4 + 1);
+  unsigned long TagID = 0;
+  unsigned short dataLength = 0, dataType = 2;
+  const TagDetails *td, *td2;
+
+  for (int i = 0; i < 100; i++) {
+    io_->readOrThrow(buf.data(), 4);
+    TagID = buf.read_uint32(0, bigEndian);
+    td = find(NikonNCTGTags, TagID);
+
+    io_->readOrThrow(buf.data(), 2);
+    dataType = Exiv2::getUShort(buf.data(), bigEndian);
```

Comment:
```suggestion
    dataType = buf.read_uint16(0, bigEndian);
```
Please could you replace this throughout the file?

Reply:
Done :heavy_check_mark: